packages/muelu/research/mmayr/composite_to_regions/src/SetupRegionSmoothers_def.hpp

```diff
@@ -105,8 +105,46 @@ void jacobiSetup(RCP<Teuchos::ParameterList> params,
     // extract inverse of diagonal from matrix
     diag[j] = VectorFactory::Build(regionGrpMats[j]->getRowMap(), true);
     regionGrpMats[j]->getLocalDiagCopy(*diag[j]);
-    diag[j]->elementWiseMultiply(SC_ONE, *diag[j], *regionInterfaceScaling[j], SC_ZERO); // ToDo Does it work to pass in diag[j], but also return into the same variable?
+    diag[j]->elementWiseMultiply(SC_ONE, *regionInterfaceScaling[j], *diag[j], SC_ZERO); // ToDo Does it work to pass in diag[j], but also return into the same variable?
     diag[j]->reciprocal(*diag[j]);
+
+    // RCP<Teuchos::FancyOStream> out = Teuchos::fancyOStream(Teuchos::rcpFromRef(std::cout));
+    // regionInterfaceScaling[j]->describe(*out, Teuchos::VERB_EXTREME);
+    // diag[j]->describe(*out, Teuchos::VERB_EXTREME);
+  }
+
+  params->set<Teuchos::Array<RCP<Vector> > >("jacobi: inverse diagonal", diag);
+}
+
+/*! \brief performs Jacobi setup
```

Comment:
The name kinda says that.  Also, Doxygen documention should go in the declarations header, not in the definitions header.

Reply:
Oh there is no declaration header, this is only dirty experimental code...